symfony/plugins/orangehrmTimePlugin/service/CustomerService.php

```diff
@@ -17,68 +17,97 @@
  * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
  * Boston, MA  02110-1301, USA
  */
-class CustomerService extends BaseService {
 
-    private $customerDao;
+namespace OrangeHRM\Time\Service;
+
+use Exception;
+use OrangeHRM\Entity\Customer;
+use OrangeHRM\Time\Dao\CustomerDao;
+use OrangeHRM\Time\Dto\CustomerSearchFilterParams;
+use OrangeHRM\Core\Exception\ServiceException;
+
+
+class CustomerService
+{
+
+    /**
+     * @var CustomerDao|null
+     */
+    private ?CustomerDao $customerDao = null;
 
     /**
      * Construct
      */
-    public function __construct() {
+    public function __construct()
```

Comment:
no need to create Dao in the constructor. Dao initialization handled in getter. remove this constructor

Reply:
done